test/Oscoin/Test/P2P/Disco.hs

```diff
@@ -0,0 +1,234 @@
+module Oscoin.Test.P2P.Disco
+    ( tests
+    -- shut up, weeder!
+    -- , prop_multicast
+    -- , prop_staticPeers
+    -- , prop_dnsSD
+    )
+where
+
+import           Oscoin.Prelude hiding (retry)
+
+import           Oscoin.P2P.Disco (Options(..), withDisco)
+import qualified Oscoin.P2P.Disco.MDns as MDns
+import           Oscoin.P2P.Types (Network, readNetwork, readNodeAddr)
+
+import           Control.Retry
+import qualified Data.Set as Set
+import qualified Network.DNS as DNS
+import           Network.Socket
+                 ( AddrInfo(..)
+                 , AddrInfoFlag(..)
+                 , HostAddress
+                 , HostAddress6
+                 , PortNumber
+                 , SockAddr(..)
+                 , SocketType(Datagram)
+                 , bind
+                 , close
+                 , defaultHints
+                 , defaultProtocol
+                 , getAddrInfo
+                 , socket
+                 , socketPort
+                 , tupleToHostAddress
+                 , tupleToHostAddress6
+                 )
+import           Network.Socket.ByteString (recvFrom, sendAllTo)
+
+import           Hedgehog
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import           Test.Tasty (TestTree, testGroup)
+import           Test.Tasty.Hedgehog (testProperty)
+
+tests :: TestTree
+tests = testGroup "Disco"
```

Comment:
I can see we have quite a fair amount of hand-rolled parsers which can be easily turned into roundtrip specs of the form `renderFoo . parseFoo == id`. What do you think?

Reply:
2d3333b05822d1815b06c15dd41196708cbad1e1